File: lua5.4/matematica/3096.lua
local function kamenetsky(n)
    if n < 0 then
        return 0
    end

    if n <= 1 then
        return 1
    end

    local x = ((n * math.log10(n/math.exp(1))) +
                (math.log10(2*math.pi*n) / 2.0))

    return math.floor(x) + 1
end

local n = io.read("*n")
print(kamenetsky(n))

File: dart/matematica/3096.dart
import 'dart:io';
import 'dart:math';

int kamenetsky(double n) {
  if (n < 0) {
    return 0;
  }

  if (n <= 1) {
    return 1;
  }

  double x = ((n * (log(n / e) / ln10) ) + ( (log(2 * pi * n) / ln10) / 2.0));

  return (x.floor() + 1);
}

void main() {
  double n = double.parse(stdin.readLineSync()!);
  print(kamenetsky(n));
}

